src/olympia/files/tests/test_utils.py

```diff
@@ -1216,65 +1217,122 @@ def test_raises_error_for_absolute_path(self):
             utils.SafeTar.open(filename)
 
 
-class TestArchiveMemberValidator(TestCase):
-    # We cannot easily test `archive_member_validator` so let's test
-    # `_validate_archive_member_name_and_size` instead.
+class TestArchiveMemberValidatorZip(TestCase):
+    def _fake_archive_member(self, filename, filesize):
+        info = zipfile.ZipInfo(filename)
+        info.file_size = filesize
+        return info
 
     def test_raises_when_filename_is_none(self):
+        info = self._fake_archive_member('', 123)
+        info.filename = None
         with pytest.raises(utils.InvalidArchiveFile):
-            utils._validate_archive_member_name_and_size(None, 123)
+            utils.archive_member_validator(info)
+
+    def test_unsupported_compression_method(self):
+        info = self._fake_archive_member('foo', 123)
+        info.compress_type == zipfile.ZIP_DEFLATED
+        # ZIP_DEFLATED works.
+        utils.archive_member_validator(info)
+        # ZIP_STORED works.
+        info.compress_type = zipfile.ZIP_DEFLATED
```

Comment:
This is also setting `compress_type` to `zipfile.ZIP_DEFLATED`.

Reply:
:man_facepalming:

Reply:
ead5ccc56acda3b3a22a6b845d9e23a9c1fe91c2